launch/launch/__init__.py

```diff
@@ -75,8 +78,11 @@ def __init__(self, cmd, name, output_handler, exit_handler, env=None):
 
     def send_signal(self, signal):
         if self.transport:
+            self.task_state.signals_received.append(signal)
             self.transport.send_signal(signal)
 
     def terminate(self):
-        if self.transport:
+        if os.name != 'nt':
+            self.send_signal(signal.SIGINT)
```

Comment:
I still think that setting the result of the future from `on_stdout_lines` in the matching handler so that the run loop wakes up from yield is necessary. Then I think I need to add some logic so that launch waits a short time before sending SIGTERM to processes that have already received SIGINT (the signals are recorded in the `signals_received` list in TaskState).